build.gradle.kts

```diff
@@ -53,24 +61,140 @@ tasks.named<Wrapper>("wrapper").configure { distributionType = Wrapper.Distribut
 // Pass environment variables:
 //    ORG_GRADLE_PROJECT_sonatypeUsername
 //    ORG_GRADLE_PROJECT_sonatypePassword
-// OR in ~/.gradle/gradle.properties set
-//    sonatypeUsername
-//    sonatypePassword
-// Call targets:
-//    publishToSonatype
-//    closeAndReleaseSonatypeStagingRepository
-nexusPublishing {
-  transitionCheckOptions {
-    // default==60 (10 minutes), wait up to 120 minutes
-    maxRetries = 720
-    // default 10s
-    delayBetween = java.time.Duration.ofSeconds(10)
+// Gradle targets:
+//    publishAggregateMavenCentralDeployment
+//    (zipAggregateMavenCentralDeployment to just generate the single, aggregated deployment zip)
+// Ref: Maven Central Publisher API:
+//    https://central.sonatype.org/publish/publish-portal-api/#uploading-a-deployment-bundle
+mavenCentralPortal {
+  credentials {
+    username.value(provider { System.getenv("ORG_GRADLE_PROJECT_sonatypeUsername") })
+    password.value(provider { System.getenv("ORG_GRADLE_PROJECT_sonatypePassword") })
   }
-  repositories {
-    // see https://central.sonatype.org/publish/publish-portal-ossrh-staging-api/#configuration
-    sonatype {
-      nexusUrl.set(uri("https://ossrh-staging-api.central.sonatype.com/service/local/"))
-      snapshotRepositoryUrl.set(uri("https://central.sonatype.com/repository/maven-snapshots/"))
+
+  deploymentName = "${project.name}-$version"
+
+  // publishingType
+  //   AUTOMATIC = fully automatic release
+  //   USER_MANAGED = user has to manually publish/drop
+  publishingType =
+    if (System.getenv("CI") != null) PublishingType.AUTOMATIC else PublishingType.USER_MANAGED
```

Comment:
Aka: in CI, release if successful. When run manually, do not publish automatically.